### ISSUE TITLE
Expose storage account and event hub resource id in azurerm_purview_account

### ISSUE THREAD

**Issue by Noel-Jones** [hashicorp/terraform-provider-azurerm#14814]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

In order to create the ingest Private Linked Endpoints that can be created via the portal using Terraform it is necessary to know the id of the managed resources; storage account and event hub namespace. The name, and therefore the id of both contains a random element which it is not possible to determine from the current attributes returned by the resource (you can actually work out the namespace from the kafka connection strings but it is very messy).

What I want to be able to do is automate the PLE creation in the following manner (i'm using a module to create the PLE but it should be sufficient to show the requirement):

```hcl
resource "azurerm_purview_account" "purview_account" {
  name                        = "purview_test"
  resource_group_name         = "myRG"
  location                    = "uksouth
  public_network_enabled      = false
  managed_resource_group_name = "purviewRG"
}

# Create Private Linked Endpoint for the ingest storage account
# There is no reference available for the storage account. Terraform does not allow for searching and in any case the deny assignment on the SA
# prevents a data resource from functioning. At this time we have no choice but to hardcode the resource id :-/
module "ple_ingest_sa" {
  for_each            = toset(["blob", "queue"])
  source              = "modules/privatelinkendpoint"
  endpoint_name       = "ple-${azurerm_purview_account.purview_account.name}-${each.key}"
  location            = "uksouth"
  resource_group_name = "networks"
  subnet_id           = "default_subnet"
  manual_connection   = false

  connection_name          = "ple-${azurerm_purview_account.purview_account.name}-${each.key}"
  ### HERE WE NEED THE EVENT STORAGE ACCOUNT ID
  ### private_link_resource_id = "/subscriptions/00000-00000-0000-00000/resourceGroups/rg-purview-sandbox/providers/Microsoft.Storage/storageAccounts/scanuksouthhhsryhy"
  private_link_resource_id = azurerm_purview_account.purview_account.managed_resources.storage_account_id
  sub_resources            = [each.key]
  tags                     = null
}

# Create Private Linked Endpoint for the ingest event hub
# There is no reference available for the namespace but we can fudge it. This is really an intellectual exercise best avoided :-/
module "ple_ingest_eh" {
  for_each            = toset(["namespace"])
  source              = "modules/privatelinkendpoint"
  endpoint_name       = "ple-${azurerm_purview_account.purview_account.name}-${each.key}"
  location            = "uksouth"
  resource_group_name = "networks"
  subnet_id           = "default_subnet"
  manual_connection   = false

  connection_name          = "ple-${azurerm_purview_account.purview_account.name}-${each.key}"
  ### HERE WE NEED THE EVENT NAMESPACE ID
  #### private_link_resource_id = "/subscriptions/00000-00000-00000-00000/resourceGroups/rg-purview-sandbox/providers/Microsoft.EventHub/namespaces/${nonsensitive(split(".", split("/", azurerm_purview_account.purview_account.atlas_kafka_endpoint_primary_connection_string)[2])[0])}"
  private_link_resource_id = azurerm_purview_account.purview_account.managed_resources.event_hub_namespace_id 
  sub_resources            = [each.key]
}

# Create Private Linked Endpoints for the account and portal
# Not shown since it is already possible
```

### New or Affected Resource(s)

* azurerm_purview_account

### Potential Terraform Configuration

No change to the resource call but needs to return additional attributes perhaps within a managed resources block. These can be per the API

```hcl
managed_resources:
  event_hub_namespace_id - the managed event hub namespace resource identifier
  resource_group_id - the managed resource group resource identifier. This resource group will host resource dependencies for the account
  storage_account_id - he managed storage account resource identifier
```

[Note that the name of the storage account, while potentially useful, is insufficient (were it available) due to the deny assignment against the storage account which prevents a data resource looking up the storage account details.]

### References

 *  https://docs.microsoft.com/en-us/rest/api/purview/accounts/get#managedresources

**Comment by Noel-Jones** [comment]
Hi @tombuildsstuff, I am new to developing Terraform resources but I put together a change that is working in my sandbox. Simple question is what now? I know the documentation would need updating. I guess I should do a test case. Appreciate any pointers. Thx.

```Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply":

  # azurerm_purview_account.purview_account has changed
  ~ resource "azurerm_purview_account" "purview_account" {
        id                                               = "/subscriptions/00000-0-0-00000/resourceGroups/rg-noel-jones/providers/Microsoft.Purview/accounts/purview-sandbox"
      + managed_resources                                = [
          + {
              + event_hub_namespace = "/subscriptions/00000-0-0-00000/resourceGroups/rg-purview-sandbox/providers/Microsoft.EventHub/namespaces/Atlas-00000-0-0-00000"
              + resource_group      = "/subscriptions/00000-0-0-00000/resourceGroups/rg-purview-sandbox"
              + storage_account     = "/subscriptions/00000-0-0-00000/resourceGroups/rg-purview-sandbox/providers/Microsoft.Storage/storageAccounts/scanuksouthhjswedh"
            },
        ]
        name                                             = "purview-sandbox"
        tags                                             = {}
        # (10 unchanged attributes hidden)
    }
```

**Comment by katbyte** [comment]
closed by #14814
